src/main/java/org/jenkinsci/maven/plugins/hpi/ValidateMojo.java

```diff
@@ -1,24 +1,33 @@
 package org.jenkinsci.maven.plugins.hpi;
 
+import com.google.common.collect.Sets;
 import hudson.util.VersionNumber;
 import org.apache.maven.plugin.MojoExecutionException;
 import org.apache.maven.plugin.MojoFailureException;
 import org.apache.maven.plugins.annotations.LifecyclePhase;
 import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+
+import java.io.IOException;
+import java.util.Set;
+import java.util.jar.Attributes;
+import java.util.jar.JarFile;
 
 /**
  * Make sure that we are running in the right environment.
  *
  * @author Kohsuke Kawaguchi
  */
-@Mojo(name = "validate", defaultPhase = LifecyclePhase.VALIDATE)
-public class ValidateMojo extends AbstractJenkinsMojo {
+@Mojo(name = "validate", defaultPhase = LifecyclePhase.VALIDATE, requiresDependencyResolution = ResolutionScope.RUNTIME)
+public class ValidateMojo extends AbstractHpiMojo {
     public void execute() throws MojoExecutionException, MojoFailureException {
-        if (!isMustangOrAbove())
+        if (!isMustangOrAbove()) {
             throw new MojoExecutionException("JDK6 or later is necessary to build a Jenkins plugin");
-
-        if (new VersionNumber(findJenkinsVersion()).compareTo(new VersionNumber("1.419.99"))<=0)
+        }
+        if (new VersionNumber(findJenkinsVersion()).compareTo(new VersionNumber("1.419.99"))<=0) {
             throw new MojoExecutionException("This version of maven-hpi-plugin requires Jenkins 1.420 or later");
```

Comment:
needs to be updated in a separate pull request. We have a decision to drop support of pre-Java 8 Jenkins core versions from development tools, and IMHO it makes sense to apply it here. Can be done in a follow-up PR